xds/src/main/java/io/grpc/xds/XdsClientImpl.java

```diff
@@ -786,25 +804,34 @@ private void handleRdsResponse(DiscoveryResponse rdsResponse) {
 
     // Notify the ConfigWatcher if this RDS response contains the most recently requested
     // RDS resource.
-    if (clusterName != null) {
+    if (routes != null) {
       if (rdsRespTimer != null) {
         rdsRespTimer.cancel();
         rdsRespTimer = null;
       }
-      logger.log(XdsLogLevel.INFO, "Found cluster name: {0}", clusterName);
-      ConfigUpdate configUpdate = ConfigUpdate.newBuilder().setClusterName(clusterName).build();
+
+      // Found clusterName in the in-lined RouteConfiguration.
+      String clusterName = routes.get(routes.size() - 1).routeAction.get().cluster;
+      if (!enablePathMatching) {
+        logger.log(XdsLogLevel.INFO, "Found cluster name: {0}", clusterName);
```

Comment:
cluster name is always queried, why only logged when path matching is enabled?

Reply:
cluster name is logged when path matching is _NOT_ enabled, which keeps the existing behavior.

When path matching is enabled, it logs the whole list of routes. In this case, the default route (last in the list) may not be a cluster, it could be weighted clusters.

Reply:
But cluster name (`routes.get(routes.size() - 1).getRouteAction().get().getCluster()`) is always queried and `clusterName` in `ConfigUpdate` is always set. Is that information used when path matching is enabled? If it is, it should be logged.

Reply:
When path matching is enabled:
In the weighted cluster case (), `getCluster()` returns an empty string, it does not make sense to log the empty string. The xds-routing lb will only use the list of routes in configUpdate.

When path matching is not enabled (existing behavior):
The `getCluster()` of the default (last) route should always return a valid cluster name, in that case we log it. The cds lb will only use the clusterName in configUpdate.